apis/Google.Cloud.Spanner.Data/Google.Cloud.Spanner.Data/SpannerConnectionStringBuilder.cs

```diff
@@ -456,7 +456,7 @@ private SpannerConnectionStringBuilder(SpannerConnectionStringBuilder other) : t
             CredentialOverride = other.CredentialOverride;
             SessionPoolManager = other.SessionPoolManager;
             EnvironmentVariableProvider = other.EnvironmentVariableProvider;
-            ConversionOptions = other.ConversionOptions;
+            // Note: ConversionOptions is populated by the connection string.
```

Comment:
Wondering if you considered implementing the `ConversionOptions` property with a similar approach to all the rest, i.e. `GetValueOrDefault`. I understand that `ConversionOptions` is a composition of several connection string parameters, so implementation of `GetValueOrDefault` wouldn't be exactly the same for `ConversionOptions` as it is for the rest, but we wouldn't have to make these exceptions here, and I believe it would be less prone to bugs in the future.

Reply:
I looked into this... ConversionOptions is fetched a *lot* - I started looking at caching the constructed value along with the strings it was constructed from. I believe it's simpler to still populate it as we see the relevant properties.
(I'm actually *removing* the exception here, and instead treating it as being like any of the other properties which is derived from the connection string handling. It was already that way, but removing the assignment in the constructor handles it more uniformly.)

Reply:
Yes, I figured that it being used a lot might be a factor in wanting to have it permanently populated. It's fine then.